src/main/java/eu/dissco/core/digitalspecimenprocessor/service/HandleService.java

```diff
@@ -30,62 +64,261 @@
+
+    // 44: referent
+    fdoRecord.add(new HandleAttribute(FIELD_IDX.get(REFERENT), REFERENT,
+        TO_BE_FIXED.getBytes(StandardCharsets.UTF_8)));
+
+    // 210: objectType
+    fdoRecord.add(
+        new HandleAttribute(FIELD_IDX.get(OBJECT_TYPE), OBJECT_TYPE,
+            "Digital Specimen".getBytes(StandardCharsets.UTF_8)));
+
+    return fdoRecord;
+  }
+
+  private List<HandleAttribute> fillFdoRecordSpecimenAttributes(DigitalSpecimen digitalSpecimen) {
+    List<HandleAttribute> fdoRecord = new ArrayList<>();
+    // 42: referentName
+    fdoRecord.add(
+        new HandleAttribute(FIELD_IDX.get(REFERENT_NAME), REFERENT_NAME,
+            getAttributeFromDigitalSpecimen(digitalSpecimen, ODS_PREFIX + "specimenName",
+                UNKNOWN).getBytes(StandardCharsets.UTF_8)));
+
+    // 200: SpecimenHost
+    fdoRecord.add(
+        new HandleAttribute(FIELD_IDX.get(SPECIMEN_HOST),
+            SPECIMEN_HOST,
+            getAttributeFromDigitalSpecimen(digitalSpecimen, ODS_PREFIX + "organisationId",
+                UNKNOWN).getBytes(StandardCharsets.UTF_8)));
+
+    // 201: Specimen Host Name
+    fdoRecord.add(
+        new HandleAttribute(FIELD_IDX.get(SPECIMEN_HOST_NAME), SPECIMEN_HOST_NAME,
+            getAttributeFromDigitalSpecimen(digitalSpecimen, ODS_PREFIX + "organisationName",
+                UNKNOWN).getBytes(StandardCharsets.UTF_8)));
+
+    // 202: PrimarySpecimenObjectId
+    fdoRecord.add(
+        new HandleAttribute(FIELD_IDX.get(PRIMARY_SPECIMEN_OBJECT_ID),
+            PRIMARY_SPECIMEN_OBJECT_ID,
+            digitalSpecimen.physicalSpecimenId().getBytes(
+                StandardCharsets.UTF_8)));
+
+    // 203: primarySpecimenObjectIdType
+    fdoRecord.add(
+        new HandleAttribute(FIELD_IDX.get(PRIMARY_SPECIMEN_OBJECT_ID_TYPE),
+            PRIMARY_SPECIMEN_OBJECT_ID_TYPE,
+            getAttributeFromDigitalSpecimen(digitalSpecimen, ODS_PREFIX + "physicalSpecimenIdType",
+                UNKNOWN).getBytes(StandardCharsets.UTF_8)));
+
+    // 204: primarySpecimenObjectIdName
+    var collectionId = getAttributeFromDigitalSpecimen(digitalSpecimen,
+        ODS_PREFIX + "physicalSpecimenCollection", "");
+    if (!collectionId.isEmpty()) {
+      var idName = ("Local identifier for collection " + collectionId).getBytes(
+          StandardCharsets.UTF_8);
+      fdoRecord.add(new HandleAttribute(FIELD_IDX.get(PRIMARY_SPECIMEN_OBJECT_ID_NAME),
+          PRIMARY_SPECIMEN_OBJECT_ID_NAME, idName));
+    }
+
+    // 206: otherSpecimenIds
+    var otherSpecimenIds = getAttributeFromDigitalSpecimen(digitalSpecimen, "dwca:id", "");
+    if (!otherSpecimenIds.isEmpty()) {
+      fdoRecord.add(new HandleAttribute(FIELD_IDX.get(OTHER_SPECIMEN_IDS), OTHER_SPECIMEN_IDS,
+          otherSpecimenIds.getBytes(
+              StandardCharsets.UTF_8)));
+    }
+
+    // 211: livingOrPreserved
+    var livingOrPreserved = getUnharmonisedAttributeFromDigitalSpecimen(digitalSpecimen,
```

Comment:
Sorry probably did not explain this carefully enough. The basis of record could be a set of different types (preservedSpecimen, rockSpecimen, livingSpecimen etc...). I would check here for a `dwc:basisOfRecord` or `abcd:recordBasis` get the value, uppercase it and trim it, then check if it is `LIVINGSPECIMEN` if not then it is preserved. See a check in the translator service as well -> DwcaService lin 280

Reply:
[Jira](https://naturalis.atlassian.net/browse/DD-454)